models/RuleAction.php

```diff
@@ -43,10 +45,27 @@ class RuleAction extends Model
         'notification_rule' => [NotificationRule::class, 'key' => 'rule_host_id'],
     ];
 
-    public function triggerAction($params)
+    public function triggerAction($params, $scheduled=true)
     {
         try {
-            $this->getActionObject()->triggerAction($params);
+            $actionObject = $this->getActionObject();
+
+            // Check if action is muted in which case we don't proceed sending a notification
+            if (method_exists($actionObject, 'isMuted') && $actionObject->isMuted()) {
```

Comment:
Correct me if I'm wrong, but as far as I can see the ActionBase class doesn't implement this functionality; RuleAction inherits from Model so it can be extended but $actionObj is not an instance of RuleAction but a subclass of ActionBase.

Reply:
Never mind, you're correct - didn't read it closely enough :) It extends ExtensionBase, not Extendable; my mistake